### PR TITLE
chore: call feature configs use case in  slow sync

### PR DESCRIPTION
**Pull request by MohammedMokresh**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

after some suggestion , we moved the get feature configs to be called in the sync instead of we call it from the app 

### Solutions

call the get feature configs from the slow sync manager and save the values to be used later 

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SlowSyncWorker.kt [hunk 1]
```diff
@@ -18,6 +19,7 @@ class SlowSyncWorker(
         kaliumLogger.d("Sync: Starting SlowSync")
 
         val result = try{
+            userSessionScope.getRemoteFeatureConfigsStatusAndPersist
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SlowSyncWorker.kt:22]: Shouldn't we invoke `()` on this?

`userSessionScope.getRemoteFeatureConfigsStatusAndPersist()` ?

Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SlowSyncWorker.kt:22]: Also, could you please add a TODO to move the feature flags stuff out of the SlowSync later? I'll explain in the review comment.

**Review by vitorhugods — COMMENTED** [review]
I think this is a great initial step, as by doing this we have a unified and orchestrated way of fetching remote configs every so often.

But it would be awesome if in the future we can take these away from Sync itself, and just wait to do this "after sync is complete" or similar.

For extra context, slowSync is currently happening everytime now, but in the future it should happen only once every week or similar.

Again: for now, it's great.
But please add a TODO so we can move this call to somewhere outside SlowSync when Sync is fully matured.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SlowSyncWorker.kt (hunk 1 above)
Comment (MohammedMokresh) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SlowSyncWorker.kt]: yeah , already add it , my bad

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigEventEnums.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+@Suppress("EnumNaming")
+enum class FeatureConfigStatus {
+    enabled,
+    disabled
+}
+
+@Suppress("EnumNaming")
+enum class FeatureConfigName {
+    fileSharing
+}
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigEventEnums.kt:13]: Same here.
Network should be responsible for mapping it to a strongly-typed enum. Not `:logic`:

```kotlin
@Serializable
enum class FeatureConfigNameDTO {
    @SerialName("fileSharing") FILE_SHARING
}
```

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigEventResponse.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.wire.kalium.logic.data.featureConfig
+
+
+import com.wire.kalium.network.api.featureConfigs.ConfigsStatus
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class FeatureConfigEventResponse(
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigEventResponse.kt:9]: This class is not being used.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/GetRemoteFileSharingStatusAndPersistUseCase.kt [hunk 4]
```diff
@@ -5,69 +5,64 @@ import com.wire.kalium.logic.configuration.UserConfigRepository
...
 import com.wire.kalium.network.exceptions.KaliumException
 import com.wire.kalium.network.exceptions.isNoTeam
 
 /**
  * This use case is to get the file sharing status of the team management settings from the server and
  * save it in the local storage (in Android case is shared preference)
  */
-interface GetRemoteFeatureConfigStatusAndPersistUseCase {
-    suspend operator fun invoke(): GetFeatureConfigStatusResult
+internal interface GetRemoteFeatureConfigStatusAndPersistUseCase {
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/GetRemoteFileSharingStatusAndPersistUseCase.kt:18]: Can we rename this to `SyncFeatureConfigsUseCase` and `SyncFeatureConfigsUseCaseImpl` so it matches the other UseCases being called during SlowSync?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsFileSharingEnabledUseCase.kt [hunk 5]
```diff
@@ -11,27 +12,27 @@ import com.wire.kalium.logic.kaliumLogger
...
 class IsFileSharingEnabledUseCaseImpl(
     private val userConfigRepository: UserConfigRepository
 ) : IsFileSharingEnabledUseCase {
 
-    override operator fun invoke(): Boolean =
+    override operator fun invoke(): FileSharingEntity =
         userConfigRepository.isFileSharingEnabled().fold({
             when (it) {
                 StorageFailure.DataNotFound -> {
                     kaliumLogger.e("Data not found in IsFileSharingEnabledUseCase")
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsFileSharingEnabledUseCase.kt:26]: We could add this same log to `FileSharingStatusFLowUseCase`, so we have the info in both places.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/FileSharingStatusFlowUseCase.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * This use case is to get a flow of  file sharing status of the team settings from the local storage
+ * so we can use it to show and hide things on the screen
+ */
+
+interface FileSharingStatusFlowUseCase {
+    operator fun invoke(): Flow<FileSharingEntity>
+}
+
+class FileSharingStatusFlowUseCaseImpl(private val userConfigRepository: UserConfigRepository) : FileSharingStatusFlowUseCase {
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/FileSharingStatusFlowUseCase.kt:20]: As agreed a few collectives ago, can we rename this to `ObserveFileSharingStatusUseCase` and `ObserveFileSharingStatusUseCaseImpl`?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsFileSharingEnabledUseCase.kt [hunk 7]
```diff
@@ -11,27 +12,27 @@ import com.wire.kalium.logic.kaliumLogger
...
-                    false
+                    FileSharingEntity(null, null)
                 }
                 is StorageFailure.Generic -> {
                     kaliumLogger.e("Storage Error : ${it.rootCause} in IsFileSharingEnabledUseCase")
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsFileSharingEnabledUseCase.kt:30]:
```suggestion
                    kaliumLogger.e("Storage Error : ${it.rootCause} in IsFileSharingEnabledUseCase", it.rootCause)
```

So the whole StackTrace gets logged too.

**Review by vitorhugods — DISMISSED** [review]
Nice job! I think it all works and is quite a nice finding and development you did in the past days.

Just some suggestions to make it a bit cleaner and easier to expand later on with other features.

**Review by MohamadJaara — APPROVED** [review]
lgtm 🚀

**Pull request merged**